### DISCUSSION TITLE
Get paired/connected devices

### DISCUSSION THREAD

**Discussion by TheLastGimbus** [bluez/bluer#120] [Q&A]
Hi 👋👋

I'm new to this library, and i wanted to quickly get currently paired devices, and print their names and mac. But this is currently impossible? Or am I dumb?

**Comment by surban** [comment]
You can use [`Adapter::device_addresses`](https://docs.rs/bluer/latest/bluer/struct.Adapter.html#method.device_addresses) to enumerate all currently known devices. This should include all paired and connected devices. From there see the [`discover_devices`](https://github.com/bluez/bluer/blob/master/bluer/examples/discover_devices.rs) examples on how to print device information.
